src/main/java/qtlog/LogParser/LogParser.java

```diff
@@ -0,0 +1,257 @@
+import qtlog.shared.RawLogDTO;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class LogParser implements ILogParser{
+    private int agentCount;
+    private int skillCount;
+    @Override
+    public RawLogDTO giveLogToParse(FileDTO fileInfo) {
+        byte[] fileContent;
+        ByteArrayInputStream byteArrayStream;
+        Path filePath = Path.of(fileInfo.getFilepath());
+
+        if(logIsZipped(fileInfo.getFilename())){
+            try {
+                byte[] unzippedLog = unzipLog(fileInfo.getFilepath());
+                byteArrayStream = new ByteArrayInputStream(unzippedLog);
+                return parseLog(byteArrayStream);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        } else {
+            try {
+                fileContent = Files.readAllBytes(filePath);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+            byteArrayStream = new ByteArrayInputStream(fileContent);
+            return parseLog(byteArrayStream);
+        }
+        return new RawLogDTO();
+    }
+
+    private boolean logIsZipped(String filename){
+        return filename.charAt(filename.length() - 5) == 'z';
+    }
+
+    private byte[] unzipLog(String filepath) throws IOException{
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        byte[] buffer = new byte[1024];
+
+        try (ZipInputStream zipInputStream = new ZipInputStream(new FileInputStream(filepath))) {
+            ZipEntry zipEntry = zipInputStream.getNextEntry();
+
+            while (zipEntry != null) {
+                int len;
+                while ((len = zipInputStream.read(buffer)) > 0) {
+                    byteArrayOutputStream.write(buffer, 0, len);
+                }
+
+                zipEntry = zipInputStream.getNextEntry();
+            }
+        }
+        return byteArrayOutputStream.toByteArray();
+    }
+
+    private RawLogDTO parseLog(ByteArrayInputStream byteStream){
+        RawLogDTO tempLogDTO = new RawLogDTO();
+
+        try {
+            //parse log
+            EvtcHeader header = getHeader(byteStream);
+            agentCount = this.getInt(byteStream);
+            ArrayList<EvtcAgent> agents = getAgentInfo(byteStream);
+            skillCount = this.getInt(byteStream);
+            ArrayList<EvtcSkill> skills = getSkillInfo(byteStream);
+            ArrayList<EvtcCombatEvent> cbtevts = getCombatEventsInfo(byteStream);
+
+            //fill relevant data for further processing
+            //get time passed
+            long timePassed = cbtevts.get(cbtevts.size() - 1).getTime() - cbtevts.get(0).getTime();
```

Comment:
> BUT theres smth weird with the time, I always get 7-10s more than what EL shows me
That's because you're checking the duration of the log and not for boss-death/reward-chest/etc